app/controllers/case_requests_controller.rb

```diff
@@ -1,27 +1,66 @@
 class CaseRequestsController < ApplicationController
+  rescue_from GlimrApiClient::Case::InvalidCaseNumber, with: :case_not_found
+
   def new
-    @case_request = CaseRequest.new(nil, nil)
+    @case_request = CaseRequest.new
   end
 
   def create
-    case_reference, confirmation_code = case_request_params
-    @case_request = CaseRequest.new(case_reference, confirmation_code)
+    @case_request = CaseRequest.new(case_request_params)
 
-    if @case_request.valid?
+    respond_to do |format|
+      format.html do
+        process_html(@case_request)
+      end
+      format.json do
+        process_json(@case_request)
+      end
+    end
+  end
+
+  def show
+    @case_request = CaseRequest.find_by_id(params[:id])
+    if @case_request
       @case_request.process!
```

Comment:
@digitalronin you're right, it shouldn't.  It was the simplest (read laziest 🙄) way to deal with the fact that 'CaseRequest#fees' is delegated to the glimr API client, which makes a has_many association more work.  I'll refactor.